docs/core_modules/data_modules/storage/root.md

```diff
@@ -1 +1,68 @@
-# Storage
\ No newline at end of file
+# Storage
+
+## Concept
+
+LlamaIndex provides a high-level interface for ingesting, indexing, and querying your external data.
+
+Under the hood, LlamaIndex also supports swappable **storage components** that allows you to customize:
+
+- **Document stores**: where ingested documents (i.e., `Node` objects) are stored,
+- **Index stores**: where index metadata are stored,
+- **Vector stores**: where embedding vectors are stored.
+
+The Document/Index stores rely on a common Key-Value store abstraction, which is also detailed below.
+
+LlamaIndex supports persisting data to any storage backend supported by [fsspec](https://filesystem-spec.readthedocs.io/en/latest/index.html). 
+We have confirmed support for the following storage backends:
+
+- Local filesystem
+- AWS S3
+- Cloudflare R2
+
+
+![](/_static/storage/storage.png)
+
+## Usage Pattern
+
+To use these storage abstractions, you need to define a `StorageContext` object:
+
+```python
+from llama_index.storage.docstore import SimpleDocumentStore
+from llama_index.storage.index_store import SimpleIndexStore
+from llama_index.vector_stores import SimpleVectorStore
+from llama_index.storage import StorageContext
+
+# create storage context using default stores
+storage_context = StorageContext.from_defaults(
+    docstore=SimpleDocumentStore(),
+    vector_store=SimpleVectorStore(),
```

Comment:
Do we want to mention that most vector store integrations store the entire index in the vector store?

I feel like this is a super missed point by most people. They try to use persist() and load_index_from_storage, instead of realizing the persistence happens automatically and they should use from_vector_store() to "reload" the index?

Fine with having this in another page, but I think it's not clear enough judging by how often people are confused about this

Reply:
This is a good point, will do!